src/lib/storage/storage_manager.cpp

```diff
@@ -30,123 +34,149 @@ void StorageManager::add_table(const std::string& name, std::shared_ptr<Table> t
   }
 
   // Create table statistics and chunk pruning statistics for added table.
+
   table->set_table_statistics(TableStatistics::from_table(*table));
   generate_chunk_pruning_statistics(table);
 
-  _tables.emplace(name, std::move(table));
+  if (table_iter == _tables.end())
+    _tables.emplace(name, std::move(table));
+  else
+    _tables[name] = std::move(table);
 }
 
 void StorageManager::drop_table(const std::string& name) {
-  const auto num_deleted = _tables.erase(name);
-  Assert(num_deleted == 1, "Error deleting table " + name + ": _erase() returned " + std::to_string(num_deleted) + ".");
+  const auto table_iter = _tables.find(name);
+  Assert(table_iter != _tables.end() && table_iter->second, "Error deleting table. No such table named '" + name + "'");
+
+  // The concurrent_unordered_map does not support concurrency-safe erasure. Thus, we simply replace the table.
+  _tables[name] = nullptr;
 }
 
 std::shared_ptr<Table> StorageManager::get_table(const std::string& name) const {
-  const auto iter = _tables.find(name);
-  Assert(iter != _tables.end(), "No such table named '" + name + "'");
+  const auto table_iter = _tables.find(name);
+  Assert(table_iter != _tables.end() && table_iter->second, "No such table named '" + name + "'");
 
-  return iter->second;
+  return table_iter->second;
```

Comment:
This is not thread-safe.

```
Assert(table_iter != _tables.end(), ...);
auto table = table_iter->second;
Assert(table, ...);
return table;
```

should be better.

In any case, we should update the description of these methods that currently says "not thread-safe". Maybe it is a good idea to limit our guarantees to parallel operations on different tables.